python_modules/dagster/dagster/_core/storage/alembic/versions/039_d3a4c9e87af3_add_asset_policy_evaluations_table.py

```diff
@@ -0,0 +1,42 @@
+"""Add asset policy evaluations table
+
+Revision ID: d3a4c9e87af3
+Revises: 701913684cb4
+Create Date: 2023-05-09 11:50:38.931820
+
+"""
+
+import sqlalchemy as db
+from alembic import op
+from dagster._core.storage.migration.utils import has_table
+from dagster._core.storage.sql import get_current_timestamp
+from sqlalchemy.dialects import sqlite
+
+# revision identifiers, used by Alembic.
+revision = "d3a4c9e87af3"
+down_revision = "701913684cb4"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if not has_table("asset_policy_evaulations"):
```

Comment:
evaulations

Reply:
did you consider `asset_auto_materialize_evaluations`?

freshness policy is another kind of asset policy

Reply:
Do you think we'll ever have more complex freshness policies that would warrant storing the evaluation info? What about source asset observation policies? Or mayybe future asset expectations? I was thinking that this table could store future things that the asset daemon polls

Reply:
One way to think about it is: imagine a world where there was no such thing as an `AutoMaterializePolicy` – all assets were just auto-materialized and users had no control over how. I think we'd still want this table.  I.e. the "policy" is a little bit orthogonal.

I can't think of a reason we'd need this for asset observations, but I suppose we can't rule it out.  The logic for whether to trigger an asset expectation is likely to be more complex than the logic for whether to trigger an asset observation, so I could imagine wanting to record that. We could call it `asset_daemon_evaluations` to be safe? Although then what do we call the table that contains a row per evaluation (not a row per asset)?

Reply:
I'm ok with asset_daemon_evaluations. @alangenfeld convinced me that for the global record for the daemon, the access pattern matches regular sensor/schedules so we could make it another tick type inside the existing ticks table